PsfLauncher/main.cpp

```diff
@@ -13,223 +13,227 @@
+#ifndef PROCESS_CREATION_DESKTOP_APP_BREAKAWAY_DISABLE_PROCESS_TREE
+#    define PROCESS_CREATION_DESKTOP_APP_BREAKAWAY_DISABLE_PROCESS_TREE PROCESS_CREATION_DESKTOP_APPX_OVERRIDE
+#endif
 
-// Forward declaration
-void LaunchMonitorInBackground(std::filesystem::path packageRoot, const wchar_t * executable, const wchar_t * arguments, bool wait, bool asadmin);
+#ifndef PROC_THREAD_ATTRIBUTE_DESKTOP_APP_POLICY
+#    define PROC_THREAD_ATTRIBUTE_DESKTOP_APP_POLICY PROC_THREAD_ATTRIBUTE_CHILD_PROCESS_POLICY
+#endif
 
-static inline bool check_suffix_if(iwstring_view str, iwstring_view suffix)
-{
-    if ((str.length() >= suffix.length()) && (str.substr(str.length() - suffix.length()) == suffix))
-    {
-        return true;
-    }
-
-    return false;
-}
+using namespace std::literals;
 
-void Log(const char* fmt, ...)
+struct ExecutionInformation
 {
-    std::string str;
-    str.resize(256);
-
-    va_list args;
-    va_start(args, fmt);
-    std::size_t count = std::vsnprintf(str.data(), str.size() + 1, fmt, args);
-    assert(count >= 0);
-    va_end(args);
-
-    if (count > str.size())
-    {
-        str.resize(count);
-
-        va_list args2;
-        va_start(args2, fmt);
-        count = std::vsnprintf(str.data(), str.size() + 1, fmt, args2);
-        assert(count >= 0);
-        va_end(args2);
-    }
+    LPCWSTR ApplicationName;
+    std::wstring CommandLine;
+    LPCWSTR CurrentDirectory;
+};
+
+//Forward declarations
+ErrorInformation StartProcess(ExecutionInformation execInfo, int cmdShow, bool runInAppContainer);
+int launcher_main(PWSTR args, int cmdShow) noexcept;
+ErrorInformation RunScript(const psf::json_object * scriptInformation, std::filesystem::path packageRoot, LPCWSTR dirStr, int cmdShow);
+ErrorInformation GetAndLaunchMonitor(const psf::json_object * monitor, std::filesystem::path packageRoot, int cmdShow, LPCWSTR dirStr);
+ErrorInformation LaunchMonitorInBackground(std::filesystem::path packageRoot, const wchar_t * executable, const wchar_t * arguments, bool wait, bool asadmin, int cmdShow, LPCWSTR dirStr);
+static inline bool check_suffix_if(iwstring_view str, iwstring_view suffix);
+void LogStringW(const char* name, const wchar_t* value);
+void LogString(const char* name, const char* value);
+void Log(const char* fmt, ...);
+ErrorInformation StartWithShellExecute(std::filesystem::path packageRoot, std::filesystem::path exeName, std::wstring exeArgString, LPCWSTR dirStr, int cmdShow);
+bool CheckIfPowerShellIsInstalled();
 
-    str.resize(count);
-    ::OutputDebugStringA(str.c_str());
-}
-void LogString(const char* name, const char* value)
-{
-    Log("\t%s=%s\n", name, value);
-}
-void LogStringW(const char* name, const wchar_t* value)
+int __stdcall wWinMain(HINSTANCE, HINSTANCE, PWSTR args, int cmdShow)
 {
-    Log("\t%s=%ls\n", name, value);
+    return launcher_main(args, cmdShow);
 }
+
 int launcher_main(PWSTR args, int cmdShow) noexcept try
 {
+    //TEST
+    MessageBoxEx(NULL, L"In here", L"In here", 0, 0);

```

Comment:
?

Reply:
I even put //TEST to remind myself to take it out.  Good catch.